packages/core-data/src/resolvers.js

```diff
@@ -896,3 +896,34 @@ export const getRevision =
 			dispatch.receiveRevisions( kind, name, recordKey, record, query );
 		}
 	};
+
+export const getThemeFileURI =
```

Comment:
This is all highly speculative and experimental. Don't worry folks! 😅 

@andrewserong is this closer to what you were thinking?

It kinda works, but it's proving difficult to use for block props, and no idea yet how to implement for global styles.

Considering generating some temp CSS rules and injecting into the DOM 🙃 

See comment: https://github.com/WordPress/gutenberg/pull/60578/files#r1584350221

Reply:
That was the general idea I had in mind, yes, thanks for exploring it! It's a challenging one, though, as it is a bit of a messy way to handle things, but I do like the idea of us having a JS equivalent to the PHP logic, and I like how you've gone for a generic approach that could be used for things other than just the background image support 👍